### DISCUSSION TITLE
Blazor navigation breaks monaco editor?

### DISCUSSION THREAD

**Discussion by Ghevi** [microsoft/monaco-editor#4617] [Q&A]
I make this as a discussion because i'm not able to reproduce this bug on a sample app, so maybe i'm doing something wrong in the actual project where it happens. I have some screenshots, just in case you might have seen this issue before and have some tips.

When loading the page for the first time, or when doing a full refresh, the editor is fine:
![Screenshot 2024-07-26 151925](https://github.com/user-attachments/assets/695eb301-959c-41b4-bc50-4c57bda30fec)

As you may know, Blazor use internal navigation like others SPA framework, so it has a router and you can navigate without refreshing the page. This is done via NavLink or NavigationManager. If I navigate in this way to the page where i have the monaco editor, now it fails to initialize properly, without logging any error in the console:
![Screenshot 2024-07-26 160042](https://github.com/user-attachments/assets/2e86d241-223c-4757-adcf-06b0a7c21e7a)

That weid dot you see on the left seems to be a text-area element, but i don't think it's important is just that the layout is broken.
I already call dispose on the editor before navigation but even without it, it doesn't make a difference.
Sorry for not providing a repro.

**Comment by VietAnh-Ng** [comment]
I have the same issue. I noticed that the style tag with the class "monaco-colors" is not being recreated.
![Screenshot 2024-07-30 093918](https://github.com/user-attachments/assets/5d52bdf1-7976-47eb-8d4a-03996cf0ae87)

**Reply by Ghevi** [reply]
Nice find!

I could not find that class on this repository, so i did right click on the `<head>` tag , then break on subtree modifications. 
Here's the code that appends that style. In the next days i will try to see what differs on the call stack when it doesn't recreate it.
Hopefully this will be useful.

![Immagine 2024-07-30 222200](https://github.com/user-attachments/assets/997ac55a-d9f9-48e2-bffd-7bd1d4eed404)

**Reply by Ghevi** [reply]
There we go, i reproduced in a new project. It has to do with setting render modes per component instead of globally in the `HeadOutlet` and `Routes` components inside `App.razor`.
All the code that monaco editor adds after the HeadOutlet gets truncated it seems, but only if navigating back and forth.

![Screenshot 2024-07-31 120912](https://github.com/user-attachments/assets/5825feee-0c8b-4507-abed-57ef11ee8874)
![Screenshot 2024-07-31 120830](https://github.com/user-attachments/assets/cfc3f92d-fe14-4a5a-9434-a3d0bc3f4566)

**Reply by VietAnh-Ng** [reply]
I have tried it and found that the issue has been resolved. Please try adding the file 'lib/monaco-editor/min/vs/editor/editor.main.css' as shown in this example from the Monaco Editor: https://github.com/microsoft/monaco-editor/blob/main/samples/browser-script-editor/index.html
![Screenshot 2024-08-01 084910](https://github.com/user-attachments/assets/6bde94e3-6ccb-429e-b6ab-8a4c762fc2a9)

**Reply by Ghevi** [reply]
Mmm that does indeed work for the style, but what about all the other tags that does not get added? 
I miss the colors and some functionalities. If i paste them all, monaco gives many errors.

**Comment by Ghevi** [comment]
Well i opened an issue on the asp.net repo and they give me this useful answer, i'm gonna try it out, but it's not a big deal for me to use forceLoad. https://github.com/dotnet/aspnetcore/issues/57100#issuecomment-2260533045
